orttraining/orttraining/training_ops/cuda/math/softmax_grad_impl.cu

```diff
@@ -30,7 +30,7 @@ template <typename input_t, typename output_t, typename acc_t, int log2_elements
 __global__ void softmax_warp_backward(output_t* gradInput, const input_t* grad, const input_t* output, int batch_size, int stride, int element_count) {
   // WARP_SIZE and WARP_BATCH must match the return values batches_per_warp and warp_size of method warp_softmax_backward_kernel.
   constexpr int next_power_of_two = 1 << log2_elements;
-  constexpr int WARP_SIZE = (next_power_of_two < CUDA_WARP_SIZE) ? next_power_of_two : CUDA_WARP_SIZE;
+  constexpr int WARP_SIZE = (next_power_of_two < GPU_WARP_SIZE) ? next_power_of_two : GPU_WARP_SIZE;
```

Comment:
This WARP_SIZE is misleading. 
It is more like the width for a lockstep vector, maybe LOCKSTEP_SIZE or VECTOR_SIZE or VECTOR_WIDTH, or LOCKSTEP_VECTOR_WIDTH is better, instead of WARP_SIZE.
so does other warp_size below.
Also, there is a warp_reduce function, which is more like vector_reduce, or lockstep_reduce.

Reply:
agree, it is confusing. But, this PR was not created to handle this issue. I plan to have a specific PR for it. If you are interested in refactoring it, please go ahead.

Reply:
It works for using another PR. 
I just accidentally saw it, since it was in a line you also modified.